pants-plugins/experimental/ansible/lint/ansible_lint/subsystem.py

```diff
@@ -0,0 +1,76 @@
+from __future__ import annotations
+
+import os
+from typing import Iterable
+
+from pants.backend.python.goals.export import ExportPythonTool, ExportPythonToolSentinel
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.target_types import ConsoleScript
+from pants.core.util_rules.config_files import ConfigFilesRequest
+from pants.engine.rules import Rule, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.option_types import ArgsListOption, SkipOption
+from pants.util.strutil import softwrap
+
+
+class AnsibleLint(PythonToolBase):
+    options_scope = "ansible-lint"
+    name = "AnsibleLint"
+    help = softwrap(
+        """
+        The ansible-lint utility for formatting Ansible playbooks, roles and collections (https://ansible-lint.readthedocs.io/en/latest/)."""
+    )
+
+    default_version = "ansible-lint==6.3.0"
+    default_main = ConsoleScript("ansible-lint")
+
+    register_interpreter_constraints = True
+    default_interpreter_constraints = ["CPython>=3.7,<4"]
+
+    skip = SkipOption("fmt", "lint")
+    args = ArgsListOption(example="-f pep8")
+
+    # register_lockfile = True
+    # default_lockfile_resource = (
+    #     "experimental.ansible.lint.ansible_lint",
+    #     "ansible_lint.lock",
+    # )
+    # default_lockfile_path = (
+    #     "pants-plugins/experimental/ansible/lint/ansible_lint/ansible_lint.lock"
+    # )
+    # default_lockfile_url = git_url(default_lockfile_path)
+
+    def config_request(self, dirs: Iterable[str]) -> ConfigFilesRequest:
+        """ansible-lint will use the closest configuration file to the file currently being formatted, so add all of them."""
```

Comment:
Hmm, I might be mis-understanding this from the documentation - I took it to mean that the "working directory" is the one that the file being processed is in, not the directory lint is called from. That would line it up with stuff like `git`, `prettier`, `clang-format`, etc

Though, this definitely means I need to add a test for it, as there is ambiguity, that a small test would resolve pretty quickly!

```
Ansible-lint supports local configuration via a .ansible-lint or .config/ansible-lint.yml configuration files. Ansible-lint checks the working directory for the presence of this file and applies any configuration found there. The configuration file location can also be overridden via the -c path/to/file CLI flag.

When configuration file is not found in current directory, the tool will try to look for one in parent directories but it will not go outside current git repository.
```

Reply:
Confirmed you were right on this, it works the opposite of how I thought. If you run `ansible-lint` in a sub-directory, it will search its parents for a configuration file (if none are in the current dir), and everything needs to be relative to that configuration root dir. 

What I thought is that it would treat each linted file's directory as a temporary current working dir, and look for a configuration in there (ala Prettier and clang-format).

Since Pants will run from root, I guess we don't need that functionality since it seems to be useful when you're running in the subdir.